agent/ui_endpoint.go

```diff
@@ -236,13 +238,7 @@ func summarizeServices(dump structs.ServiceDump) []*ServiceSummary {
 		sum.Kind = svc.Kind
 		sum.InstanceCount += 1
 		if svc.Kind == structs.ServiceKindConnectProxy {
-			if _, ok := sum.proxyForSet[svc.Proxy.DestinationServiceName]; !ok {
-				if sum.proxyForSet == nil {
-					sum.proxyForSet = make(map[string]struct{})
-				}
-				sum.proxyForSet[svc.Proxy.DestinationServiceName] = struct{}{}
-				sum.ProxyFor = append(sum.ProxyFor, svc.Proxy.DestinationServiceName)
-			}
+			connected[structs.NewServiceID(svc.Proxy.DestinationServiceName, &svc.EnterpriseMeta)] = true
```

Comment:
We don't call `sum := getService($DESTINATION_SERVICE)` here because if the destination service isn't registered it will end up with a service summary and a row in the UI.